src/V3LinkParse.cpp

```diff
@@ -895,6 +895,15 @@ class LinkParseVisitor final : public VNVisitor {
         }
         iterateChildren(nodep);
     }
+    void visit(AstClass* nodep) override {
+	cleanFileline(nodep);
+	for (AstNode* itemp = nodep->stmtsp(); itemp; itemp = itemp->nextp()) {
+	    if (VN_IS(itemp, PackageImport)) {
+		itemp->v3error("Import statement directly within a class scope is illegal");
+	    }
```

Comment:
Instead add a visitor for PackageImport and check there what it is under that was recursed previously, e.g. `if (m_modp && !VN_IS(modp, Package) error...`.

Note this will also find imports under `module` which is presumably also not allowed either - perhaps have the test do that too?

Reply:
Instead add a visitor for PackageImport and check there what it is under that was recursed previously, e.g. `if (m_modp && !VN_IS(modp, Package)) nodep->error...`.

Note my suggestion will also find imports under `module`, which is presumably also not allowed either - perhaps have the test do that too?